File: src/MyMoviesApp.Domain/Entities/MovieDetail.cs
using MyMoviesApp.Domain.Enums;

namespace MyMoviesApp.Domain.Entities;

/// <summary>
/// Entity. Formats and DigitalRetailers can be null, otherwise they need to be a hash set.
/// </summary>
public record MovieDetail(
    int MovieId,
    string Title,
    DateOnly ReleaseDate,
    int Runtime,
    string PosterPath,
    string BackdropPath,
    string Tagline,
    string Overview
);

File: src/clients/AngularClient/src/app/movies/models/movie-detail.ts
import { Format, DigitalRetailer } from "./lookup";

export interface MovieDetail {
    tmdbId: number;
    title: string;
    releaseDate: string;
    posterPath: string | null;
    backdropPath: string | null;
    runtime: number;
    tagline: string | null;
    overview: string;
    formats: Format[];
    digitalRetailers: DigitalRetailer[];
}